egs2/binaural_librispeech/enh1/local/data.sh

```diff
@@ -0,0 +1,168 @@
+  optional argument:
+    [--min_or_max]: min (Default), max
+    [--sample_rate]: 8k (Default), 16k
+EOF
+)
+
+. ./path.sh || exit 1;
+. ./cmd.sh || exit 1;
+. ./db.sh || exit 1;
+
+cdir=$PWD
+
+# Path to the directory containing MUSAN noise
+musan=${MUSAN}
+
+# Path to the directory containing Binaural LibriSpeech
+binaural_librispeech="${BINAURAL_LIBRISPEECH}"
+binaural_librispeech_subset="horizontal_plane_front_only"
+
+# Subsets to include in data preparation
+subsets=("train-clean-100" "train-clean-360" "train-other-500" "dev-clean" "dev-other" "test-clean" "test-other")
+
+min_or_max=max
+sample_rate=16k
+num_spk=2
+
+stage=0
+stop_stage=100
+
+. utils/parse_options.sh
+
+if [ $# -ne 0 ]; then
+    echo "${help_message}"
+    exit 1;
+fi
+
+if [[ "$min_or_max" != "max" ]] && [[ "$min_or_max" != "min" ]]; then
+  echo "Error: min_or_max must be either max or min: ${min_or_max}"
+  exit 1
+fi
+if [[ "$sample_rate" != "16k" ]] && [[ "$sample_rate" != "8k" ]]; then
+  echo "Error: sample rate must be either 16k or 8k: ${sample_rate}"
+  exit 1
+fi
+
+# Simulation parameters
+log "Binaural LibriSpeech data preparation"
+
+if [ ${stage} -le 0 ] && [ ${stop_stage} -ge 0 ]; then
+    log "stage 0: Downloading BinauralLibriSpeech"
+    # Download and extract BinauralLibriSpeech data
+    if [ -z "${binaural_librispeech}" ]; then
+        if [ -d "${cdir}/downloads/BinauralLibriSpeech" ]; then
+          log "Binaural LibriSpeech data already exists in: ${cdir}/downloads/BinauralLibriSpeech."
+          log "Skipping download.."
+        else
+          mkdir -p "${cdir}/downloads/BinauralLibriSpeech"
+          log "Downloading data from HuggingFace to '${cdir}/downloads/BinauralLibriSpeech'"
+          log "Using Binaural Librispeech version: ${binaural_librispeech_subset}"
+          python local/download_binaural_librispeech.py ${binaural_librispeech_subset} "${cdir}/downloads/"
+          # Remove HF cache after download is finished.
+          rm -rf "${cdir}/downloads/.cache/huggingface/"
+          # Move "BinauralLibriSpeech" up one level to "/downloads"
+          mv "${cdir}/downloads/data/BinauralLibriSpeech" "${cdir}/downloads"
+          # Remove the now-empty "data" folder if desired
+          rmdir "${cdir}/downloads/data"
+          # Extract data
+          log "Extracting data..."
+          local/extract_tar_and_delete.sh "${cdir}/downloads/BinauralLibriSpeech"
+          log "Downloaded and extracted Binaural Librispeech data"
+        fi
+    else
+        # Copy BinauralLibriSpeech to user directory in case of permission issues.
+        log "Binaural LibriSpeech data already exists at ${binaural_librispeech}, copying to '${cdir}/downloads/BinauralLibriSpeech'"
+        mkdir -p "${cdir}/downloads/BinauralLibriSpeech"
+        log "Using Binaural Librispeech version: ${binaural_librispeech_subset}"
+        rsync -r -P "${binaural_librispeech}/${binaural_librispeech_subset}" "${cdir}/downloads/BinauralLibriSpeech"
```

Comment:
Is it necessary to copy the downloaded directory? If it is OK to make a symbolic link (via `ln -s`), it is better to replace it.